.github/workflows/pr-analysis.yml

```diff
@@ -21,33 +21,45 @@ jobs:
         uses: actions-rs/toolchain@v1
         with:
           toolchain: stable
-      - name: Install dependencies (ubuntu only)
-        if: matrix.platform == 'ubuntu-latest'
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y libgtk-3-dev webkit2gtk-4.0 libappindicator3-dev librsvg2-dev patchelf
       - name: Install app dependencies
         run: yarn install
       - name: Build app
         run: yarn build
       - name: Run linter
         run: yarn lint
       - name: Run tests
-        run: yarn test --all
+        run: yarn test a --coverage
+      - name: Upload coverage report
+        uses: actions/upload-artifact@master
+        with:
+          name: coverage
+          path: coverage/
       - name: Build Tauri native binaries
         uses: tauri-apps/tauri-action@v0
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
   sonarcloud:
     name: SonarCloud
+    if: ${{ success() }}
+    needs: build-tauri
```

Comment:
There's no need to specify `if: success` when using a `needs: previous_job`. It will [automatically stop execution if the needed job fails unless you specify no to do so](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds).